jenkins/python-setup.sh

```diff
@@ -0,0 +1,14 @@
+_prefix="${1}"
+
+pip3 install --no-cache-dir --prefix="${_prefix}" -r ${progdir}/agent/requirements.txt -r ${progdir}/server/requirements.txt -r ${progdir}/agent/test-requirements.txt -r ${progdir}/server/test-requirements.txt
+
+_pdir=${_prefix}/bin
+if [[ ":${PATH:-}:" != *":${_pdir}:"* ]]; then
+    export PATH=${_pdir}${PATH:+:${PATH}}
+fi
+
+for _pdir in $(ls -1d ${_prefix}/lib*/python3.*/site-packages 2> /dev/null) $(head -n 1 ${_prefix}/lib*/python3.*/site-packages/pbench.egg-link 2> /dev/null); do
```

Comment:
@RobertKrawitz had some comments on this in chat:

```
1) Why are you using ls -1d rather than simply globbing the expression?  If the concern is no files present, you can use shopt -s nullglob.

2) This construct is vulnerable to any spaces in the file names.  A safer way to do this would be

while read -r _pdir ; do
...
done <<< "$(ls -1d ... ; head -n 1 ...)"
```

Reply:
@ndokos, @RobertKrawitz, can python modules have spaces in their names?

Reply:
And this code was taken from https://github.com/distributed-system-analysis/pbench/blob/master/agent/profile#L27, so perhaps we could have somebody post a more thorough fixing to address the use of `ls -1d`?

Reply:
Tracked as issue #1979.